beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -0,0 +1,96 @@
+package sync
+
+import (
+	"context"
+	"sort"
+	"time"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+var processPendingBlocksPeriod = time.Duration(params.BeaconConfig().SecondsPerSlot/3) * time.Second
+
+// processes pending blocks queue on every processPendingBlocksPeriod
+func (r *RegularSync) processPendingBlocksQueue() {
+	ticker := time.NewTicker(processPendingBlocksPeriod)
+	for {
+		ctx := context.TODO()
+		select {
+		case <-ticker.C:
+			r.processPendingBlocks(ctx)
+		case <-r.ctx.Done():
+			log.Debug("p2p context is closed, exiting routine")
+			break
+
+		}
+	}
+}
+
+// processes the block tree inside the queue
+func (r *RegularSync) processPendingBlocks(ctx context.Context) error {
+	pids := r.peerIDs()
+	slots := r.sortedPendingSlots()
+
+	for _, s := range slots {
+		r.slotToPendingBlocksLock.RLock()
```

Comment:
can you use the same lock for both of these maps  ? seeing as they are read from and written together